### ISSUE TITLE
Implement offline maps

### ISSUE THREAD

**Issue by cafootitt** [OpenSRP/opensrp-client-eusm#9]
As part of this issue, we'll need to create a boundary for the district by drawing a boundary around all of the service points' GPS points in the district(s) that are part of the mission.

**Comment by bennsimon** [comment]
@cafootitt am suggesting we use commune since getting district bounding box will lead to a large number of tiles which in turn increases the download size.

**Comment by cafootitt** [comment]
16th Dec - We will go with districts instead of communes, as these seem to be downloading ok.

**Comment by cafootitt** [comment]
16 Dec 2020 build - This is looking really great!

Just one thing I noticed about how the maps are downloaded (steps below):
- Logged in as demo user
- Available maps to download were SOANIERANA IVONGO and MANANARA AVARATRA
- I selected both maps and tapped on Download button
- SOANIERANA IVONGO downloaded quite quickly and disappeared
- MANANARA AVARATRA took longer to download, but I noticed that it disappeared from the list after it started downloading; whereas with SOANIERANA IVONGO, the location showed in the list while it was downloading.
- When MANANARA AVARATRA reached about 93% download, the downloading status disappeared, yet the location didn't appear in the DOWNLOADED tab for several minutes. So there's basically a period of several minutes where the user isn't sure whether the download worked because the snack bar message with the download percentage disappeared, and the location name disappeared in both tabs.
- Apart from this small point of possible confusion for the user, the download works really well!
